app/models/workflow.rb

```diff
@@ -25,6 +26,10 @@ def external_signal?
 
   private
 
+  def cleanup_requests_and_taglinks
```

Comment:
why name the method this way? There is no actual cleanup of requests and tag links happening here. You may name it something like `validate_deletable`